### PR TITLE
Tidy up the link checking code

### PR DESCRIPTION
**Pull request by thomasleese**
Following on from @kevindew's review of #5.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@ class Application < Rails::Application
     # Application configuration should go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded.
 
+    config.eager_load_paths << "#{config.root}/app"
```
Comment (kevindew) [inline config/application.rb]: don't think we need this - probably don't want to encourage classes in root of app.

File: lib/link_check.rb [hunk 2]
```diff
@@ -3,11 +3,10 @@ class LinkCheck
 
   def initialize(uri)
     @uri = uri
-    @report = UriChecker::Report.new
+    @report = LinkCheck::UriChecker::Report.new
```
Comment (kevindew) [inline lib/link_check.rb]: these should still work without the namespace you've added in as it'll be scoped to `LinkCheck` already

File: lib/link_check/uri_checker/http_checker.rb [hunk 3]
```diff
@@ -1,55 +1,4 @@
...
-      report
-    rescue URI::InvalidURIError
-      report.errors[:uri_invalid] << "Invalid URI"
-      report
-    end
-  end
-
+module LinkCheck::UriChecker
   class HttpChecker
     INVALID_TOP_LEVEL_DOMAINS = %w(xxx adult dating porn sex sexy singles)
```
Comment (kevindew) [inline lib/link_check/uri_checker/http_checker.rb]: we should host this on one of those. `link-checker.sexy`

File: lib/link_check/uri_checker/http_checker.rb [hunk 4]
```diff
@@ -154,33 +101,41 @@ def check_google_safebrowsing
...
+        })
       end
     end
 
     def make_request(method)
       begin
         response = run_connection_request(method)
 
-        if response.status >= 300 && response.status < 400
+        if [301, 302, 303, 307, 308].include?(response.status) && !report.has_errors?
```
Comment (kevindew) [inline lib/link_check/uri_checker/http_checker.rb]: probably want to check for location header in this conditional too

File: lib/link_check/uri_checker/http_checker.rb (hunk 3 above)
Comment (thomasleese) [inline lib/link_check/uri_checker/http_checker.rb]: `link-checker.singles`

**Pull request merged**